### PR TITLE
Print error log to stderr output

### PR DESCRIPTION
**Pull request by res-life**
Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

File: dist/scripts/check-shims-revisions.sh [hunk 1]
```diff
@@ -35,15 +35,14 @@ function check-shims-revisions() {
     if [[ -f "$shim_version_path" ]] ; then
       curr_revision=$(grep "revision=" "${shim_version_path}" | cut -d'=' -f2)
       if [ -n "$pre_revision" ] && [[ "$curr_revision" != "$pre_revision" ]] ; then
-        echo
-        echo "Check Failed: git revisions between shims are not equal"
-        echo "Please check the revisions of each shim to see which one is inconsistent. Note, if building with Databricks those jars are built separately."
+        echo >&2 "Check Failed: git revisions between shims are not equal"
```
Comment (firestarman) [inline dist/scripts/check-shims-revisions.sh:38]: Missing the original empty `echo`, who will print a space line.

**Review by firestarman — COMMENTED** [review]
May I know why need to do this ?

**Comment by res-life** [comment]
The msg in stderr will be saved in the parameter  ${build-parallel-worlds.checkRevisionsErrorMsg}.  If not print to stderr, the parameter ${build-parallel-worlds.checkRevisionsErrorMsg} is empty. 

See the following:
```
        <!-- check shims revisions -->
        <exec executable="${project.basedir}/scripts/check-shims-revisions.sh"
              dir="${project.build.directory}"
              resultproperty="build-parallel-worlds.checkRevisionsExitCode"
              errorproperty="build-parallel-worlds.checkRevisionsErrorMsg"
              failonerror="false">
            <arg value="${included_buildvers}"/>
        </exec>
        <fail message="exec check-shims-revisions.sh failed, exit code is ${build-parallel-worlds.checkRevisionsExitCode}, error msg is ${build-parallel-worlds.checkRevisionsErrorMsg}">
            <condition>
                <not>
                    <equals arg1="${build-parallel-worlds.checkRevisionsExitCode}" arg2="0"/>
                </not>
            </condition>
        </fail>
```

File: dist/scripts/check-shims-revisions.sh (hunk 1 above)
Comment (res-life) [reply dist/scripts/check-shims-revisions.sh:38]: It's a empty line for show,  can be remove.

**Comment by gerashegalov** [comment]
> The msg in stderr will be saved in the parameter ${build-parallel-worlds.checkRevisionsErrorMsg}. If not print to stderr, the parameter ${build-parallel-worlds.checkRevisionsErrorMsg} is empty.

I missed that in the review too, although I thought I saw this working. Did you test it?

**Comment by res-life** [comment]
> I missed that in the review too, although I thought I saw this working. Did you test it?
Yes, tested, it works. If versions are inconsistent , the error is 
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.0.0:run (create-parallel-world) on project rapids-4-spark_2.12: An Ant BuildException has occured: The following error occurred while executing this line:
[ERROR] /home/chong/code/spark-rapids/dist/maven-antrun/build-parallel-worlds.xml:134: exec check-shims-revisions.sh failed, exit code is 1, error msg is Check Failed: git revisions between shims are not equal
[ERROR] Please check the revisions of each shim to see which one is inconsistent. Note, if building with Databricks those jars are built separately.
[ERROR] around Ant part ...<ant antfile="/home/chong/code/spark-rapids/dist/maven-antrun/build-parallel-worlds.xml" target="build-parallel-worlds" />... @ 9:127 in /home/chong/code/spark-rapids/dist/target/antrun/build-main.xml
[ERROR] -> [Help 1]

```

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Review by revans2 — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

File: dist/scripts/check-shims-revisions.sh (hunk 1 above)
Comment (res-life) [reply dist/scripts/check-shims-revisions.sh:38]: Sure, I'll update it.

**Pull request merged**